src/test/java/part1/exercise/StreamsExercise1.java

```diff
@@ -25,19 +20,26 @@
 
     @Test
     public void getAllEpamEmployees() {
-        List<Person> epamEmployees = null;// TODO all persons with experience in epam
-        throw new UnsupportedOperationException();
+        List<Person> epamEmployees = Generator.generateEmployeeList().stream()
+                .filter(e -> e.getJobHistory().stream().anyMatch(j -> j.getEmployer().equals("epam")))
+                .map(Employee::getPerson)
+                .collect(toList());
     }
 
     @Test
     public void getEmployeesStartedFromEpam() {
-        List<Person> epamEmployees = null;// TODO all persons with first experience in epam
-        throw new UnsupportedOperationException();
+        List<Person> epamEmployees = Generator.generateEmployeeList().stream()
+                .filter(e -> e.getJobHistory().stream()
+                        .findFirst()
+                        .get()

```

Comment:
Fails on empty history. Don't use `get()`.
You could use `limit(1)` instead of `findFirst`